src/service/video-service-sync-impl.js

```diff
@@ -87,17 +99,28 @@ export class VideoServiceSync {
   register(video, unusedFromV1manageAutoplay = true) {
     this.asyncImpl_.then(impl =>
       impl.register(video));
+
+    this.maybeInstallAutoplay_(video);
+
+    new VideoEntry(video);
```

Comment:
you had some factory pattern for this stuff in some other code that I remember from a while back, I really like that. worth doing a factory here?

Reply:
In this case maybe the factory function can help decouple AMP types (e.g. `BaseElement`, etc). I'm going to think about how to make the change worth something.